metsim/metsim.py

```diff
@@ -82,50 +109,57 @@ def write(self, data: dict):
                 'netcdf' : self.write_netcdf,
                 'ascii'  : self.write_ascii
                 }
-        dispatch[self.params.get('out_format', 'netcdf')](data)
+        dispatch[MetSim.params.get('out_format', 'netcdf').lower()](data)
          
 
     def init_netcdf(self):
         """Initialize the output file"""
         print("Initializing netcdf...")
-        out_file = os.path.join(self.params['out_dir'], "forcing.nc")
+        out_file = os.path.join(MetSim.params.get('out_dir', './results/'), "forcing.nc")
         self.output = Dataset(out_file, 'w')
 
         # Number of timesteps 
-        n_ts = len(pd.date_range(self.params['start'], 
-                self.params['stop'] + pd.Timedelta('1 days'), 
-                freq=self.params['time_step']+'T'))
+        n_ts = len(pd.date_range(MetSim.params['start'], 
+                MetSim.params['stop'] + pd.Timedelta('1 days'), 
+                freq=MetSim.params['time_step']+'T'))
         
         # Create dimensions
         self.output.createDimension('lat', len(self.lats))
         self.output.createDimension('lon', len(self.lons))
         self.output.createDimension('time', n_ts) 
 
         # Create output variables
-        for varname in self.params['out_vars']:
+        for varname in MetSim.params.get('out_vars', []):
             self.output.createVariable(varname, 'd', ('time', 'lat','lon'))
 
 
     def write_netcdf(self, data:dict):
-        """Write out to the output file"""
+        """Write out as NetCDF to the output file"""
         print("Writing netcdf...")
         lats = data.keys()
         for lat in lats:
             lons = data[lat].keys()
             for lon in lons:
-                for varname in self.params['out_vars']:
-                    self.output.variables[varname][:, lat, lon] = data[lat][lon][varname].values
+                for varname in MetSim.params.get('out_vars', []):
+                    self.output.variables[varname][:, lat, lon] = (
+                            data[lat][lon][varname].values)
 
 
     def write_ascii(self, data: dict):
-        """ TODO """
+        """Write out as ASCII to the output file"""
         print("Writing ascii...")
-        raise NotImplementedError 
-   
+        out_base = os.path.join(MetSim.params['out_dir'], "forcing")
+        lats = data.keys()
+        for lat in lats:
+            lons = data[lat].keys()
+            for lon in lons:
+                print(lat, lon)
```

Comment:
since this could end up being quite long, probably better to use logging module.